### ISSUE TITLE
"Conflicts=" list dropped if directive concatenates an EOF

### ISSUE THREAD

**Issue by Keeblo** [systemd/systemd#8875]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> 238

### Used distribution

> Manjaro

### In case of bug report: Expected behaviour you didn't see

> "Conflicts" directive should have been parsed.

### In case of bug report: Unexpected behaviour you saw

> When a systemd target file contains a "Conflicts" list at the end of the file and does not provide a blank line between the end of the list and the end of the file the list is silently dropped.

### In case of bug report: Steps to reproduce the problem

> This is easier to demonstrate than it is to explain.

Take the following files:

`ServiceA.service`, `ServiceB.service`, and `ServiceC.service` are identical:
```
[Unit]
Description=This service just remains on

[Service]
Restart=always
ExecStart=/home/USERNAME/Permanent/Scripts/StayRunning.sh
```

`StayRunning.sh` is:
```
#!/bin/sh

while true
	do
	sleep 3
done
```
The three services launch a script that keeps on running indefinitely.

This target, `WillTurnOff.target`, will stop all three services as we would expect:
```
[Unit]
Description=This target will turn off Service's A, B, and C
Conflicts= \
ServiceA.service \
ServiceB.service \
ServiceC.service \

```

This target, `WillNotTurnOff.target`, will **not** stop the stop the services:
```
[Unit]
Description=This target will *not* turn off Service's A, B, and C
Conflicts= \
ServiceA.service \
ServiceB.service \
ServiceC.service \
```

Notice the difference. `WillTurnOff.target` contains a blank line between the last item on the list and the end of the file.

Several things to note:
* I've tested this on my laptop running up-to-date Manjaro installation with systemd version 238 installed
* I'm guessing that systemd is concatenating the EOF onto the list since the list ends with a backslash.
* Shouldn't systemd at least spit out a warning that something is wrong?
* There's a possibility that I'm totally confused and this isn't a bug at all :-P

Thank you for all of your hard work!

**Comment by filbranden** [comment]
This is pretty much working as intended, as a line continuation expects a line after it...

If you query the properties, you'll see that line was indeed ignored:

```shell
$ systemctl show -p Conflicts WillTurnOff.target
Conflicts=shutdown.target ServiceA.service ServiceB.service ServiceC.service
```

And:

```shell
$ systemctl show -p Conflicts WillNotTurnOff.target
Conflicts=shutdown.target
```

I agree it's not ideal that this isn't logged anywhere, in the unit status or in the journal... Perhaps that part might be improved.

Or we could decide to change that. I checked with a shell script (bash) and it seems perfectly happy to have a `\` continuation at the last line of the script, that still works...

In other words, this works:

```shell
#!/bin/sh
echo abc \
def
```

And then so does this:

```shell
#!/bin/sh
echo abc \
```

Or this:

```shell
#!/bin/sh
echo abc \
def \
```

**Comment by keszybz** [comment]
I'm pretty sure we should treat the trailing line with a continuation symbol similarly to bash. This simply seems to be a bug in the parser.

**Comment by keszybz** [comment]
Damn, I wrote a nearly-identical patch but forgot to create a PR from it.
